scripts/gather_feature_toggle_state.py

```diff
@@ -134,22 +147,29 @@ def main(output_path):
 
     if os.path.isdir(output_path):
         shutil.rmtree(output_path)
+    print('Creating output directory at: {}'.format(output_path))
     os.mkdir(output_path)
 
     for ida in idas:
+        print('Gathering feature toggle state data for {}'.format(ida['app']))
         connection = create_db_connection(ida['database'])
-        cursor = connection.cursor(buffered=True)
 
-        feature_toggle_data = {}
+        try:
+            cursor = connection.cursor(buffered=True)
 
-        for table in ida['table_names']:
-            query = build_query(table)
-            cursor.execute(query)
-            tagged_data = link_headers_to_data(cursor)
+            feature_toggle_data = {}
 
-            feature_toggle_data[table] = tagged_data
+            for table in ida['table_names']:
+                query = build_query(table)
+                cursor.execute(query)
+                tagged_data = link_headers_to_data(cursor)
 
-        connection.close()
+                feature_toggle_data[table] = tagged_data
+        except connector.errors.ProgrammingError:
```

Comment:
If `create_db_connection` fails, you could also get an `AttributeError` here; not sure if you want to explicitly catch that.  I'd be inclined to just move the connection creation into the try block and add an `if connection:` condition in `finally` to catch the case where it isn't created successfully.

Reply:
I put a try/catch block in `create_db_connection` that  will exit the script if the connection isn't created successfully.